connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ConnectorConfig.java

```diff
@@ -252,4 +254,37 @@ public boolean visible(String name, Map<String, Object> parsedConfig) {
         }
     }
 
+    public static class ValidName implements ConfigDef.Validator {
+        // Define characters that are not allowed in a connector name
+        final char[] illegalCharacters = {'/', ' ', '\\', '?', '%', '"', '<', '>', '\r', '\n', '\t'};
+
+        public static ValidName validName() {
+            return new ValidName();
+        }
+
+        @Override
+        public void ensureValid(String name, Object value) {
+            String s = (String) value;
+
+            if (s == null) {
+                // This can happen during creation of the config object due to no default value being defined for the
+                // name configuration - a missing name parameter is caught when checking for mandatory parameters,
+                // thus we can ok a null value here
+                return;
+            } else if (s.isEmpty()) {
+                throw new ConfigException(name, value, "Name may not be empty");
```

Comment:
What about "Connector name may not be empty" or "Connector name must be supplied"?

Reply:
Agreed, it should be specified that the connector name is referenced here. Will change to second suggestion.